fillAssumed.py

```diff
@@ -28,9 +45,18 @@ class FillAssumed(FillAlgorithm):
     extra_items: list[Item]
     itemLists: list[list[Item]]
 
-    def __init__(self,
-                 connections: list[tuple[AreaDoor, AreaDoor]]) -> None:
-        self.connections = connections
+    def __init__(self, game) -> None:
+        self.game = game
+        self.connections = game.connections
+
+        self.forced_item_locations = []
+        # TODO ASCENT_FIX should be a CLI option with choices:
+        # force - force items into force item locations
+        # duplicate - if necessary, replace a random non-unique item with duplicate key item
+        # open - replace brown doors with pink doors to allow backtracking
+        # none - do nothing and risk softlock
+        if game.options.ascent_fix in ['force', 'duplicate']:
```

Comment:
The above comment is now I'd like to ultimately do this, with 4 options for dealing with the softlock problem.